src/NSwag.CodeGeneration.CSharp.Tests/ParameterTests.cs

```diff
@@ -272,5 +272,80 @@ public void Deep_object_properties_are_correctly_named()
             Assert.Contains("options[optionalOrder.id]=", code);
             Assert.Contains("options.OptionalOrderId", code);
         }
+
+        [Theory]
+        [InlineData(Models.CSharpControllerStyle.Abstract)]
+        [InlineData(Models.CSharpControllerStyle.Partial)]
+        public void When_file_parameter_with_AspNetCore_controller_use_IFormFile(Models.CSharpControllerStyle controllerStyle)
+        {
+            //// Arrange
+            var document = new OpenApiDocument();
+            document.Paths["foo"] = new OpenApiPathItem
+            {
+                {
+                    OpenApiOperationMethod.Get, new OpenApiOperation
+                    {
+                        Parameters =
+                        {
+                            new OpenApiParameter
+                            {
+                                Kind = OpenApiParameterKind.FormData,
+                                Name = "foo",
+                                Type = JsonObjectType.File
+                            }
+                        }
+                    }
+                }
+            };
+
+            //// Act
+            var generator = new CSharpControllerGenerator(document, new CSharpControllerGeneratorSettings
+            {
+                ControllerStyle = controllerStyle,
+                ControllerTarget = Models.CSharpControllerTarget.AspNetCore
+            });
+            var code = generator.GenerateFile();
+
+            //// Assert
+            Assert.Contains("System.Threading.Tasks.Task Foo(Microsoft.AspNetCore.Http.IFormFile foo)", code);
+        }
+
+        [Theory]
+        [InlineData(Models.CSharpControllerStyle.Abstract)]
+        [InlineData(Models.CSharpControllerStyle.Partial)]
+        public void When_multiple_file_parameter_with_AspNetCore_controller_use_IFormFile(Models.CSharpControllerStyle controllerStyle)
+        {
+            //// Arrange
+            var document = new OpenApiDocument();
+            document.Paths["foo"] = new OpenApiPathItem
+            {
+                {
+                    OpenApiOperationMethod.Get, new OpenApiOperation
+                    {
+                        Parameters =
+                        {
+                            new OpenApiParameter
+                            {
+                                Kind = OpenApiParameterKind.FormData,
+                                Name = "foo",
+                                Type = JsonObjectType.File,
+                                CollectionFormat = OpenApiParameterCollectionFormat.Multi

```

Comment:
Not 100% sure if this is correct, but setting this was the only way I could find to trigger the branch of code that generated the collection of files.

Reply:
Maybe this is something which is not yet supported with NSwag and OAS3.0...